### DISCUSSION TITLE
How can I nest interruption-proofed enumerated lists [(@)]?

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [quarto-dev/quarto-cli#7751] [Q&A]
### Description

Hello,

I am trying to wrap my head around how to control numbered lists around codeblocks. Usually, you can use `(@)` to force continued enumeration when dealing with interruptions, as detailed in the [documentation](https://quarto.org/docs/authoring/markdown-basics.html#lists). This syntax however does not seem to allow complex nesting that easily. I've experimented with a couple of things, but so far I could not figure out how to nest this kind of list successfully:

````qmd
---
format: html
---

(@) A  
(@) B  
    (@) C  

```{r}
#| eval: false
norm <- wo_outliers %>%
				group_by(Gruppe) %>%
	        dplyr::summarise(
		        statistic = shapiro.test(plant_area_plotted)$statistic,
		        p.value = shapiro.test(plant_area_plotted)$p.value
					)
```

(@) D  

```{r}
#| eval: false
norm <- wo_outliers %>%
				group_by(Gruppe) %>%
	        dplyr::summarise(
		        statistic = shapiro.test(plant_area_plotted)$statistic,
		        p.value = shapiro.test(plant_area_plotted)$p.value
					)
```

(@) E  
    (@) F  

```{r}
#| eval: false
norm <- wo_outliers %>%
				group_by(Gruppe) %>%
	        dplyr::summarise(
		        statistic = shapiro.test(plant_area_plotted)$statistic,
		        p.value = shapiro.test(plant_area_plotted)$p.value
					)
```

(@) G  
    (@) H  
        (@) I
````

This results in the following html-doc (rendered via rstudio):

![grafik](https://github.com/quarto-dev/quarto-cli/assets/80548781/44ed3528-76e1-4de5-b2c2-5cd4f8153ebd)

I would intend the lists to be enumerated as follows, in order of occurence:

```txt
1. A
2. B
2.1 C
3. D
4. E
4.1 F
5. G
5.1 H
5.1.1 I
```

Alternatively, I believe this would be a bit closer to the current method:

```txt
1. A
2. B
    1. C
3. D
4. E
    1. F
5. G
    1. H
        1. I
```
---

Currently, as far as I know one has to choose between keeping enumeration to handle disruptors and using traditional lists to handle nesting - which is bad if you need both.

Is this the case, or is there a way that I am unaware of which allows me to combine both aspects?

---

Addendum 30.11.2023 13:13:
- `docx` does not exhibit this issue, here lists are properly enumerated as outlined as expected above.
- `pdf` does exhibit the same issue as `html`.
- all formats are rendered from within Rstudio
- rendering from console set in the containing folder does not yield different results for either format.

---

Thank you,  
Sincerely,  
~Gw

**Comment by mcanouil** [comment]
`(@)` is to keep enumeration going no matter what, that's its sole purpose, this there is no way to keep enumerating no matter what but few exceptions whilst there are syntaxes for those exceptions.

See Pandoc documentation about list: <https://pandoc.org/MANUAL.html#lists> also in Quarto documentation <https://quarto.org/docs/authoring/markdown-basics.html#lists>.

If you follow the nested rules described in both documentation you get what you want:

````qmd
---
format: html
---

1. A  
1. B  
   1. C  

      Whatever text

1. D  

   > Quotes

1. E  
   1. F  

      ```{r}
      #| eval: false
      "code"
      ```

1. G  
   1. H  
      1. I
````

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/ce33cf09-e3ab-47c4-aa44-f2aeac6439bb">

Side note: is the formatting style of your code intended because it kind of not follow any rules.
